rss2email/feed.py

```diff
@@ -833,7 +833,18 @@ def run(self, send=True):
         """
         if not self.to:
             raise _error.NoToEmailAddress(feed=self)
-        parsed = self._fetch()
+        
+        if self.section in self.config:
+            config = self.config[self.section]
+        else:
+            config = self.config['DEFAULT']
+        agent = config['user-agent']
+        if agent:
+            _feedparser.USER_AGENT = agent
+            parsed = self._fetch()
+            _feedparser.USER_AGENT = _USER_AGENT
+        else:
+            parsed = self._fetch()
```

Comment:
`Feed._fetch` is a wrapper around `feedparser.parse`, which [takes an `agent` argument][1] ([since v2][2]?).  And `Feed.load_from_config` sets all the config entries as instance attributes.  So instead of setting `_feedparser.USER_AGENT` up top and doing this little dance here, I'd probably just use:

    return f(
        self.url, self.etag, modified=self.modified,
        agent=self.user_agent or _USER_AGENT,
        **kwargs)

in `Feed._fetch`.

And we also use `_USER_AGENT` to set some headers, so those should also become `self.user_agent or _USER_AGENT`.

[1]: https://github.com/kurtmckee/feedparser/blob/5.2.1/feedparser/feedparser.py#L3818
[2]: https://github.com/kurtmckee/feedparser/commit/f043210329e1915141df23f82d2f4b5af9cd6a8e#diff-c1d3afe5805ad6b9125d4f7fefe24605R111